app/code/core/Mage/Adminhtml/Block/System/Config/Form.php

```diff
@@ -425,21 +426,27 @@ public function initFields($fieldset, $group, $section, $fieldPrefix = '', $labe
                     }
 
                     $sourceModel = Mage::getSingleton($factoryName);
+                    if (!$sourceModel) {
+                        Mage::throwException("Source model '{$factoryName}' is not found");
+                    }
                     if ($sourceModel instanceof Varien_Object) {
                         $sourceModel->setPath($path);
                     }
+                    
+                    $optionArray = [];
                     if ($method) {
                         if ($fieldType == 'multiselect') {
                             $optionArray = $sourceModel->$method();
                         } else {
-                            $optionArray = [];
                             foreach ($sourceModel->$method() as $value => $label) {
                                 $optionArray[] = ['label' => $label, 'value' => $value];
                             }
                         }
-                    } else {
+                    } elseif (method_exists($sourceModel, 'toOptionArray')) {
                         $optionArray = $sourceModel->toOptionArray($fieldType == 'multiselect');
                     }
+
+                    $optionArray ?? Mage::throwException("Missing method 'toOptionArray()' in source model '{$factoryName}'");
```

Comment:
I asked ChatGPT, it doesn't like it.
![image](https://user-images.githubusercontent.com/1106470/225788168-c9cdebe7-6500-4318-b482-ab0fef9e4c50.png)

Reply:
But, its opinion can be corrected:
![image](https://user-images.githubusercontent.com/1106470/225789243-894b6b54-7c67-454a-8a8b-2548d8f61559.png)

Reply:
Last:
![image](https://user-images.githubusercontent.com/1106470/225789810-6f3d2f73-b037-4b57-a17a-0e424a1d59a0.png)

Reply:
IMO it's confusing because it's an expression not a statement, which may give the impression that an assignment to a variable was missed. It's also not used anywhere in the code, so I prefer the longer version.

Reply:
I rewrote it in a slightly different way cause I didn't like the flow, I think it's more readable now.